### PR TITLE
#1039 Use merged and validated of version `.rultor.yml` for `merge` command.

### PR DESCRIPTION
**Pull request by xupyprmv**
#1039 I implemented validated method, that checks that merged .rultor.yml has no changes in architect and commander sections. I covered `validated` method by unit tests. I added one puzzle (to implement `merged` method, write IT for `validated` method and use it.

This is a copy of https://github.com/yegor256/rultor/pull/1064 that cannot be reopened. That's why I created new one.

### CODE REVIEW

**Comment by original-brownbear** [comment]
@xupyprmv while I just give this a quick read to be safe :)
You you please add the closed PR to this PR's description and also explain the situation as to why we have two PRs here in the issue ?
Thanks!

File: src/main/java/com/rultor/profiles/Profiles.java [hunk 1]
```diff
@@ -73,33 +78,107 @@ public Profile fetch(final Talk talk) throws IOException {
...
-                        xml.xpath(
-                            "//request/args/arg[@name='fork']/text()"
-                        ).get(0)
-                    )
-                ),
+            final Profile master = new GithubProfile(
+                    new TalkIssues(Profiles.github(), xml).get().repo()
+            );
+            profile = this.merged(
+                master,
```
Comment (original-brownbear) [inline src/main/java/com/rultor/profiles/Profiles.java]: @xupyprmv the variable `master` is redundant here, please inline this real quick :)

**Comment by original-brownbear** [comment]
@xupyprmv found one small thing, not sure if this was in the other PR and I just missed it (can't see it anymore now :/) or somehow new. Either way please fix it :) Thanks!

**Comment by xupyprmv** [comment]
@original-brownbear I modified PR description and fixed code. Great thanks for help. Please continue.

**Comment by original-brownbear** [comment]
@rultor merge please :)

**Comment by original-brownbear** [comment]
@xupyprmv thanks for the good job :)

**Comment by original-brownbear** [comment]
@rultor deploy

**Pull request merged**